### DISCUSSION TITLE
Best way to load child/parent roles and permissions?

### DISCUSSION THREAD

**Discussion by Sean-OShea** [spatie/laravel-permission#2081] [Q&A]
Hi everyone,

My use case is that i have a hierarchy of users. And sometimes i need to check the permission of a child or parent user. I've been surprised that none of those works (please correct me if i'm wrong).

- eager load by default on the User model : `protected $with = ['roles.permissions', 'permissions'];`
- eager load on the relationship method
```
    public function parent(){
        return $this->belongsTo('App\Models\User')->with(['roles.permissions', 'permissions']);
    }
```

Because i don't want to have to explicitly eager load every time i get my user's parent, here is the method i use :

```
    /**
     * @return User|null
     */
    public function getParent()
    {
        $parent = $this->parent;
        $parent?->load('roles.permissions', 'permissions');
        return $parent;
    }
```

Am i missing something? I feel like it must be a common use case but i haven't found a discussion on that specific parent/child relationship. Thanks everyone and have a nice day.
Sean.

Edit : i must specify that i'm also using the team feature, but a user hierarchy always share the same team_id (and it is set on each request with a middleware for the authenticated user as described in the documentation).

**Comment by zenmediasocial** [comment]
Hey there - I have developed a very elegant way to handle this - basically Zanzibar style permissions management using nothing more than Spatie Permissions and Teams.  You can read my article on Substack - or shoot me a message - and I can go over it in detail. Link in Bio. It's really great - not only does it grant permissions and handle this kind of tiered relationship but it also handles negative permissions on a per-account level using groups as well.  The best part - it keeps Users table clean - and works great with Memcache et al.
